tomato/tests/tests_viewset_Tomato.py

```diff
@@ -573,22 +571,137 @@ def test_statistics_tomatoes_year(self):
+            [
+                {
+                    'data': [],
+                    'label': 'number_of_tomato'
+                }
+            ],
+        )
+
+    def test_statistics_tomatoes_year(self):
+        """
+        Test we can get tomatoes statistics of current year
+        """
+        today = timezone.now()
+        last_year = today - timedelta(days=366)
+        limit_last_year = last_year + timedelta(days=1)
+
+        end = today.strftime('%Y-%m-%dT%H:%M:%SZ')
+        start = limit_last_year.strftime('%Y-%m-%dT%H:%M:%SZ')
+        self.client.force_authenticate(user=self.user)
+
         Tomato.objects.create(user=self.user, number_of_tomato=5)
         Tomato.objects.create(user=self.admin, number_of_tomato=4)
         Tomato.objects.create(user=self.user, number_of_tomato=7)
         Tomato.objects.all().update(acquisition_date=last_year)
 
-        t1 = Tomato.objects.create(user=self.user, number_of_tomato=15)
-        t2 = Tomato.objects.create(user=self.admin, number_of_tomato=23)
-        t3 = Tomato.objects.create(user=self.user, number_of_tomato=45)
+        t1 = Tomato.objects.create(
+            user=self.user,
+            number_of_tomato=15,
+            acquisition_date=today - timedelta(days=1),
+        )
+        t2 = Tomato.objects.create(
+            user=self.admin,
+            number_of_tomato=23,
+            acquisition_date=today - timedelta(days=2),
+        )
+        t3 = Tomato.objects.create(
+            user=self.user,
+            number_of_tomato=45,
+            acquisition_date=today - timedelta(days=3),
+        )
         current_entries = [
-            t1.number_of_tomato, t2.number_of_tomato, t3.number_of_tomato]
+            t1.number_of_tomato,
+            t2.number_of_tomato,
+            t3.number_of_tomato,
+        ]
         user_entries = [
-            t1.number_of_tomato, t3.number_of_tomato]
+            t1.number_of_tomato,
+            t3.number_of_tomato,
+        ]
+
+        response = self.client.get(
+            reverse('tomato-statistics'),
+            data={
+                'start_date': start,
+                'end_date': end
+            },
+            content_type='application/json',
+        )
+        result = response.json()
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_200_OK,
+            response.content
+        )
+
+        self.assertEqual(result['totals']['global'], sum(current_entries))
+        self.assertEqual(result['totals']['user'], sum(user_entries))
+        self.assertEqual(
+            result['graph']['labels'],
+            ['2023-02-01T00:00:00-0500'],
+        )
+        self.assertEqual(
+            result['graph']['datasets'],
+            [
+                {
+                    'label': 'number_of_tomato',
+                    'data': [
+                        {
+                            'x': '2023-02-01T00:00:00-0500',
+                            'y': 83.0
```

Comment:
shouldn't we put variable data based on init var instead of hardcoded result ? strgfy datetime expected and decimal() ?